### PR TITLE
Update UI for components-react FilterBuilder

### PR DESCRIPTION
**Pull request by morganadrales**
## Changes
Before:
image

After:
image

API changes include removal of nesting, as we will be able to support "IN" and "BETWEEN" operators instead.
Note: This branch is a fork of https://github.com/iTwin/appui/pull/576 where the sizes have been removed until iTwinUI's ButtonGroup is implemented for the EnumButtonGroupEditor.

## Testing
Updated tests to reflect the removal of nesting. Maintained the package's 100% test coverage.

### CODE REVIEW

File: ui/components-react/src/test/filter-builder/FilterBuilderRuleGroup.test.tsx [hunk 1]
```diff
@@ -130,122 +101,119 @@ describe("PropertyFilterBuilderRuleGroupRenderer", () => {
...
+      PropertyFilterRuleGroupOperator.Or
+    );
   });
 
-  it("dispatches add rule group event when button is clicked", () => {
+  it("Toggles operator 'Or' to 'And'", async () => {
     const actions = new PropertyFilterBuilderActions(sinon.spy());
-    const { getByTestId } = renderWithContext(
-      <PropertyFilterBuilderRuleGroupRenderer {...defaultProps} />,
+    rootGroup.operator = PropertyFilterRuleGroupOperator.Or;
```
Comment (raplemie) [inline ui/components-react/src/test/filter-builder/FilterBuilderRuleGroup.test.tsx:135]: On the other PR, there was a remark about mutating rootGroup, for this you could make a shallow copy of the shared group and change the copy operator.

File: ui/components-react/src/test/filter-builder/FilterBuilderRuleGroup.test.tsx [hunk 2]
```diff
@@ -130,122 +101,119 @@ describe("PropertyFilterBuilderRuleGroupRenderer", () => {
...
+    const props: PropertyFilterBuilderRuleGroupRendererProps = {
+      group: rootGroup,
```
Comment (raplemie) [inline ui/components-react/src/test/filter-builder/FilterBuilderRuleGroup.test.tsx:137]: On the other PR, there was a remark about mutating `rootGroup`, I think it was missed in the branch.
```suggestion
    rootGroup.operator = PropertyFilterRuleGroupOperator.Or;
    const props: PropertyFilterBuilderRuleGroupRendererProps = {
      group: { ...rootGroup, operator: PropertyFilterRuleGroupOperator.Or },
```

**Review by raplemie — CHANGES_REQUESTED** [review]
Add a NextVersion.md entry for these changes, outside of the few comments, we seem to be pretty done with this!

File: common/changes/@itwin/core-react/morgan-filter-builder_2024-01-12-17-01.json
[the remark this answers is not in the document]
Comment (morganadrales) [reply common/changes/@itwin/core-react/morgan-filter-builder_2024-01-12-17-01.json:1]: Should I add a 4.9.0 entry? Or what version will this be making it into?

[the remark this answers is not in the document]
Comment (raplemie) [reply common/changes/@itwin/core-react/morgan-filter-builder_2024-01-12-17-01.json:1]: There is a file named "NextVersion.md", just edit it in place, the numbering will be handled by the tooling.

File: ui/components-react/src/test/filter-builder/FilterBuilderRuleGroup.test.tsx [hunk 3]
```diff
@@ -130,122 +101,123 @@ describe("PropertyFilterBuilderRuleGroupRenderer", () => {
...
-          id: "id",
-          groupId: "parentGroupId",
-          operator: PropertyFilterRuleGroupOperator.And,
-          items: [],
-        }}
-      />,
+    const rootGroupCopy = Object.create(rootGroup);
+    rootGroupCopy.operator = PropertyFilterRuleGroupOperator.Or;
+    const props: PropertyFilterBuilderRuleGroupRendererProps = {
+      group: rootGroupCopy,
```
Comment (saskliutas) [inline ui/components-react/src/test/filter-builder/FilterBuilderRuleGroup.test.tsx:175]:
```suggestion
    const props: PropertyFilterBuilderRuleGroupRendererProps = {
      group: { ...rootGroup, operator: PropertyFilterRuleGroupOperator.Or  },
```

File: ui/components-react/src/test/filter-builder/FilterBuilderRuleGroup.test.tsx [hunk 4]
```diff
@@ -130,122 +101,123 @@ describe("PropertyFilterBuilderRuleGroupRenderer", () => {
...
-  it("dispatches add rule group event when button is clicked", () => {
+  it("Toggles operator 'Or' to 'And'", async () => {
     const actions = new PropertyFilterBuilderActions(sinon.spy());
-    const { getByTestId } = renderWithContext(
-      <PropertyFilterBuilderRuleGroupRenderer {...defaultProps} />,
+    //make shallow copy to avoid altering rootGroup that is used in other tests
+    const rootGroupCopy = Object.create(rootGroup);
+    rootGroupCopy.operator = PropertyFilterRuleGroupOperator.Or;
+    const props: PropertyFilterBuilderRuleGroupRendererProps = {
+      group: rootGroupCopy,
```
Comment (saskliutas) [inline ui/components-react/src/test/filter-builder/FilterBuilderRuleGroup.test.tsx:139]:
```suggestion
    const props: PropertyFilterBuilderRuleGroupRendererProps = {
      group: { ...rootGroup, operator: PropertyFilterRuleGroupOperator.Or },
```

**Review by saskliutas — CHANGES_REQUESTED** [review]

**Review by saskliutas — APPROVED** [review]

File: docs/changehistory/NextVersion.md [hunk 5]
```diff
@@ -9,6 +9,11 @@ Table of contents:
 
 ## @itwin/components-react
 
+## Changes
```
Comment (raplemie) [inline docs/changehistory/NextVersion.md:12]: Add the `changes` section in the table of content.

**Review by raplemie — APPROVED** [review]
With this one last thing, good job, then done with this!

**Pull request merged**